.github/workflows/backup-paas-database.yml

```diff
@@ -10,31 +10,110 @@ on:
           - staging
           - sandbox
           - production
+        required: true
 
 jobs:
   backup:
-    name: Backup
+    name: Backup (${{ github.event.inputs.environment }})
     runs-on: ubuntu-latest
+    environment:
+      name: ${{ github.event.inputs.environment }}
 
     steps:
-      - name: Setup cf cli
+      - name: Setup cf cli (staging)
+        if: github.event.inputs.environment == 'staging'
         uses: DFE-Digital/github-actions/setup-cf-cli@master
         with:
           CF_USERNAME: ${{ secrets.GOVPAAS_STAG_USERNAME }}
           CF_PASSWORD: ${{ secrets.GOVPAAS_STAG_PASSWORD }}
           CF_SPACE_NAME: early-careers-framework-staging
           INSTALL_CONDUIT: true
 
+      - name: Setup cf cli (sandbox)
+        if: github.event.inputs.environment == 'sandbox'
+        uses: DFE-Digital/github-actions/setup-cf-cli@master
+        with:
+          CF_USERNAME: ${{ secrets.GOVPAAS_SAND_USERNAME }}
+          CF_PASSWORD: ${{ secrets.GOVPAAS_SAND_PASSWORD }}
+          CF_SPACE_NAME: early-careers-framework-sandbox
+          INSTALL_CONDUIT: true
+
+      - name: Setup cf cli (production)
+        if: github.event.inputs.environment == 'production'
+        uses: DFE-Digital/github-actions/setup-cf-cli@master
+        with:
+          CF_USERNAME: ${{ secrets.GOVPAAS_PROD_USERNAME }}
+          CF_PASSWORD: ${{ secrets.GOVPAAS_PROD_PASSWORD }}
+          CF_SPACE_NAME: early-careers-framework-prod
+          INSTALL_CONDUIT: true
+
       - name: Setup postgres client
         uses: DFE-Digital/github-actions/install-postgres-client@master
 
-      - name: Backup database
+      - name: Backup database (staging, sandbox)
+        if: github.event.inputs.environment == 'staging' || github.event.inputs.environment == 'sandbox'
         run: |
-          cf conduit ecf-postgres-staging -- pg_dump -E utf8 --clean --compress=1 --if-exists --no-owner --no-privileges --verbose -f backup.sql.gz
+          cf conduit ecf-postgres-${{ github.event.inputs.environment }} -- pg_dump -E utf8 --clean --compress=1 --if-exists --no-owner --no-privileges --verbose -f backup.sql.gz
+
+      # this is temporary while we test restoring production using the backup, re-merge it with
+      # the step above when we're ready to test for real
+      - name: Backup database (production)
+        if: github.event.inputs.environment == 'production'
+        run: |
+          cf conduit ecf-postgres-production-backup-restore -- pg_dump -E utf8 --clean --compress=1 --if-exists --no-owner --no-privileges --verbose -f backup.sql.gz
```

Comment:
maybe we should use different filenames, as in `sandbox-backup` etc. for each env?

Reply:
discussed in person: not necessary as artifact scoped to workflow

Reply:
Leaving for now as the artefacts are scoped by workflow. Will change them if our test run fails.